src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java

```diff
@@ -17,7 +18,9 @@ public class DriverManager {
      * @param driver Set the driver as current.
      */
     public synchronized void setCurrentDriver(Job2dDriver driver) {
-        currentDriver = driver;
+        WorkspaceFeature.updateDriver(driver);
+
+        currentDriver = WorkspaceFeature.getWorkspaceManager();
```

Comment:
More DIP, and OCP! You shouldn't apply direct changes to DriverManager. Use observer pattern (see changePublisher.notifyObservers();), add this behaviour when setCurrentDriver is called.